DNN Platform/Library/Entities/Modules/Settings/SettingsRepository.cs

```diff
@@ -180,6 +204,11 @@ private T Load(CacheItemArgs args)
                     settingValue = (string)ctlModule.ModuleSettings[mapping.FullParameterName];
                 }
 
+                if (attribute.IsSecure)
+                {
+                    settingValue = Security.FIPSCompliant.DecryptAES(settingValue, Config.GetDecryptionkey(), Host.Host.GUID);

```

Comment:
Should we add a try-catch here? Decryption will bomb if the string is not decryptable. This could be because a developer has changed the IsSecure setting on a property for an update of his/her module. Then the minute the new code runs, it find the old value and will break quite nasty. I wouldn't mind adding a try-catch so the value is passed as-is if it can't be decrypted.

Reply:
:+1: sounds good to me